packages/react-reconciler/src/ReactFiberHydrationContext.old.js

```diff
@@ -556,6 +572,16 @@ function popHydrationState(fiber: Fiber): boolean {
   return true;
 }
 
+function hasUnhydratedTailNodes() {
```

Comment:
I like this name but just to add clarification for future readers.

There's a lot of terminology in the code base already that refers to "dehydrated" things. There's a subtle difference between "dehydrated" and "unhydrated". The dehydrated things are in a pending state yet to be hydrated. The unhydrated was ones that failed after hydration. So I think it makes sense to use "unhydrated" here.